driver/docker/docker.go

```diff
@@ -48,6 +50,12 @@ func (d *Driver) AddConfigurationOptions(opts ...ConfigurationOption) {
 	d.dockerConfigurationOptions = append(d.dockerConfigurationOptions, opts...)
 }
 
+// GetContainerConfig returns the container configuration and host configuration
+// used by the driver during container exec
+func (d *Driver) GetContainerConfig() (*container.Config, *container.HostConfig) {
```

Comment:
Almost nit-picking, but I'm a bit on the fence with this getter. I understand that it may be easier to use, but I think it's not a common golang pattern to return two unexposed fields at once 🤔 
WDYT @carolynvs ?

Reply:
I see 3 different configuration variables that would be useful to query for unit tests: driver config (`d.config`), container config (`d.containerCfg`), and host config (`d.containerHostCfg`). Right now what is confusing is that this function is named like an accessor for just ContainerConfig, but it returns both container and host config. 

I recommend making 3 accessors, one for each, that returns copies instead of mutatable pointers to the config. Otherwise we open up another pattern for setting config on the driver that we didn't intend and more likely give people a way to write buggy tests. Splitting them up allows the tests to grab just the config variable they needed, without any extra processing.

Reply:
Good call, thank you.  Split into two functions for accessing container config and container host config.  (If/when we desire a driver config accessor, we can add one in the future).   Also brought in [copystructure](https://godoc.org/github.com/mitchellh/copystructure) to assist in returning copies.  WDYT?